### ISSUE TITLE
[BUG]dask_cudf index merge fails with MultiIndex object has no attribute dtype

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#4982]
**Describe the bug**
We currently seem to fail when doing a multi-index merge with `dask_cudf`. 

**Steps/Code to reproduce bug**
```python
>>> import dask_cudf
>>> import cudf
>>> df_2 = cudf.DataFrame({'g_1': [1,1,2,2,3,1,2],'g_2': [1,1,2,2,3,1,2],'c': [4,5,6,10,11,23,11]}).set_index(['g_1','g_2'])
>>> df_1 = cudf.DataFrame({'g_1': [1,1,2,2,3,1,2],'g_2': [1,1,2,2,3,1,2],'b': [4,5,6,10,11,23,11]}).set_index(['g_1','g_2'])
>>> ddf_1 = dask_cudf.from_cudf(df_1,npartitions=1)
>>> ddf_2 = dask_cudf.from_cudf(df_2,npartitions=1)
>>> ddf_2.merge(ddf_1)
```

```python

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vjawa/conda/envs/rapids_cudf_21_ucx_14_apr_pack_frames_ucx/lib/python3.7/site-packages/dask_cudf/core.py", line 151, in merge
    return super().merge(other, on=on, shuffle="tasks", **kwargs)
  File "/home/vjawa/conda/envs/rapids_cudf_21_ucx_14_apr_pack_frames_ucx/lib/python3.7/site-packages/dask/dataframe/core.py", line 3936, in merge
    shuffle=shuffle,
  File "/home/vjawa/conda/envs/rapids_cudf_21_ucx_14_apr_pack_frames_ucx/lib/python3.7/site-packages/dask/dataframe/multi.py", line 489, in merge
    right_index=right_index,
  File "/home/vjawa/conda/envs/rapids_cudf_21_ucx_14_apr_pack_frames_ucx/lib/python3.7/site-packages/dask/dataframe/multi.py", line 254, in merge_indexed_dataframes
    kwargs["empty_index_dtype"] = meta.index.dtype
AttributeError: 'MultiIndex' object has no attribute 'dtype'

```

**Environment overview (please complete the following information)**
```python
# packages in environment at 
/home/vjawa/conda/envs/rapids_cudf_21_ucx_14_apr_pack_frames_ucx:
cudf                      0.14.0a200421         py37_3439    rapidsai-nightly
dask-cudf                 0.14.0a200421         py37_3439    rapidsai-nightly
```

**Comment by VibhuJawa** [comment]
> I thought Dask doesn't support MultiIndexes in general where I'd expect the `from_cudf` call to fail here.

Did not know that but that seems like the case. 

https://github.com/dask/dask/issues/1493

**Comment by rjzamora** [comment]
> I thought Dask doesn't support MultiIndexes in general..

Right - Distributed `MultiIndex` is not supported in `dask.dataframe` or `dask_cudf`

> ..I'd expect the from_cudf call to fail here.

Interesting - It seems that `from_cudf` is equivalent to `from_pandas`.  The tasks are even labeled "from_pandas-*"...

```python
import dask_cudf
import cudf

df = cudf.DataFrame({"a": list(range(100))})
ddf = dask_cudf.from_cudf(df, npartitions=1)
dict(ddf.dask)
```
**Output**:
```

{('from_pandas-93f1e8890111e48ffdf165e208a0a931',
  0):      a
 0    0
 1    1
 2    2
 3    3
 4    4
 ..  ..
 95  95
 96  96
 97  97
 98  98
 99  99
 
 [100 rows x 1 columns]}
```

The `from_pandas` definition **does** raise an error for `pd.MultiIndex`, but does not check for any cudf types.

**Comment by kkraus14** [comment]
Fixed by #4983
